BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/SelfDiagnostics/SelfDiagnosticsConfigParser.cs

```diff
@@ -31,11 +36,149 @@ internal class SelfDiagnosticsConfigParser
         // In theory the variable won't be access at the same time because worker thread first Task.Delay for a few seconds.
         private byte[] configBuffer;
 
+        /// <summary>
+        /// Represents the location for App Insights to parse user-defined self-diagnostics settings.
+        /// </summary>
+        internal enum ParseLocation
+        {
+            /// <summary>
+            /// Parse self-diagnostics settings from enviornment variable(s).
+            /// </summary>
+            EnviornmentVariable,
+
+            /// <summary>
+            /// Parse self-diagnostics settings from tje JSON file.
+            /// </summary>
+            ConfigJson,
+        }
+
         public bool TryGetConfiguration(out string logDirectory, out int fileSizeInKB, out EventLevel logLevel)
+        {
+            if (TryGetConfigFromEnvrionmentVariable(out logDirectory, out fileSizeInKB, out logLevel))
+            {
+                // self-diagnostics settings passed in via enviornment variables has higher precedence than
+                // self-diagnostcis settings passed in via JSON file.
+                return true;
+            }
+
+            return this.TryGetConfigFromJsonFile(ref logDirectory, ref fileSizeInKB, ref logLevel);
+        }
+
+        internal static bool TryGetConfigFromEnvrionmentVariable(out string logDirectory, out int fileSizeInKB, out EventLevel logLevel)
         {
             logDirectory = null;
             fileSizeInKB = 0;
             logLevel = EventLevel.LogAlways;
+
+            if (!TryParseLogDirectory(ParseLocation.EnviornmentVariable, Environment.GetEnvironmentVariable(LogDirectory), out logDirectory))
+            {
+                return false;
+            }
+
+            if (!TryParseFileSize(ParseLocation.EnviornmentVariable, Environment.GetEnvironmentVariable(FileSize), out fileSizeInKB))
+            {
+                return false;
+            }
+
+            UpdateFileSizeToBeWithinLimit(ref fileSizeInKB);
+
+            if (!TryParseLogLevel(ParseLocation.EnviornmentVariable, Environment.GetEnvironmentVariable(LogLevel), out logLevel))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        internal static bool TryParseLogDirectory(ParseLocation location, string val, out string logDirectory)
```

Comment:
Do we need an enum `ParseLocation`?
You could call this method, if env variable is not null. This will reduce the changes to these existing methods.

Reply:
Other option is to use overloaded methods.

Reply:
One other option is to control the behavior through a new property or field instead of passing the hard-coded value as first parameter and validating within the method.

Reply:
I've removed the Enum. Please checkout the current design. Thank you!